python/pyspark/sql/functions.py

```diff
@@ -2205,6 +2205,10 @@ def udf(f=None, returnType=StringType()):
         rows that do not satisfy the conditions, the suggested workaround is to incorporate the
         condition logic into the functions.
 
+    .. note:: Users can't rely on short-curcuit evaluation of boolean expressions to execute
```

Comment:
Looks okay but how about combining this comment with the one above and making it shorter if possible? Like .. udfs don't support conditional expression or short curcuiting and it ends up with being executed all internally. If it depends on ..., workaround blabla and blabla.

Reply:
Just a little bit worried of overwhelming users with maybe too much information although it might be worth.

Reply:
Ok.